src/components/activity/ActivityDirectiveForm.svelte

```diff
@@ -231,19 +236,25 @@
     await applyPresetToActivity(selectedPreset, numOfUserChanges);
   }
 
-  function onChangeFormParameters(event: CustomEvent<FormParameter>) {
+  async function onChangeFormParameters(event: CustomEvent<FormParameter>) {
     const { detail: formParameter } = event;
     const { arguments: argumentsMap, id } = activityDirective;
     const newArguments = getArguments(argumentsMap, formParameter);
     if ($plan) {
-      effects.updateActivityDirective(
+      if (onChangeActivityArgumentsController) {
+        onChangeActivityArgumentsController.abort();
```

Comment:
Might be nice to filter out abort errors in the effect with `if ((e as Error).name !== 'AbortError') ...` so that they don't log to the user error and dev consoles.